### PR TITLE
Rewrite TLS code to return booleans instead of ints

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
@jimis @vpodzime please review

**Comment by olehermanse** [comment]
https://ci.cfengine.com/view/all/job/pr-pipeline/50/

File: cf-serverd/server_tls.c [hunk 1]
```diff
@@ -184,7 +184,7 @@ void ServerTLSDeInitialize()
  *
  * @return -1 in case of error, 1 otherwise.
  */
-int ServerTLSPeek(ConnectionInfo *conn_info)
+bool ServerTLSPeek(ConnectionInfo *conn_info)
```
Comment (jimis) [inline cf-serverd/server_tls.c]: Don't forget to adjust the comments in all the functions you change the signature.

**Comment by olehermanse** [comment]
All green in jenkins

File: cf-serverd/server_tls.c (hunk 1 above)
Comment (olehermanse) [reply cf-serverd/server_tls.c]: Thanks, fixed.

**Pull request merged**